src/doc/tarpl/vec-deref.md

```diff
@@ -0,0 +1,40 @@
+% Deref
+
+Alright! We've got a decent minimal ArrayStack implemented. We can push, we can
+pop, and we can clean up after ourselves. However there's a whole mess of functionality
+we'd reasonably want. In particular, we have a proper array, but none of the slice
+functionality. That's actually pretty easy to solve: we can implement `Deref<Target=[T]>`.
+This will magically make our Vec coerce to and behave like a slice in all sorts of
+conditions.
+
+All we need is `slice::from_raw_parts`.
```

Comment:
Does this handle an empty vector properly?

(I know it does, but it seems maybe worth mentioning it? I dunno.)